core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1500,6 +1508,8 @@ public void forceMerge(final boolean flush, int maxNumSegments, boolean onlyExpu
                 if (flush) {
                     if (tryRenewSyncCommit() == false) {
                         flush(false, true);
+                    } else {
+                        refresh("renew sync commit"); // we have to refresh both searchers here to ensure we release unreferenced segments.
```

Comment:
Isn't tryRenewSyncCommit already doing it in this case?

Reply:
see my comment above sorry for the noise